### ISSUE TITLE
Add termination bound in 

### ISSUE THREAD

**Issue by wbhart** [wbhart/flint2#865]
Add a termination bound in _fmpq_mat_can_solve_multi_mod (which is for non-square and singular matrices) as we have in _fmpq_mat_solve_multi_mod (for square, non-singular matrices).

See the file fmpq_mat/can_solve_multi_mod.c

**Comment by thofma** [comment]
Is this code correct without the termination bound?

**Comment by wbhart** [comment]
Yes, it is. It's an early termination bound that I'm suggesting.

I have half of it sorted out, but didn't get to figuring out the other half yet.
